app/lib/hyrax/hirmeos/client.rb

```diff
@@ -1,14 +1,15 @@
 # frozen_string_literal: true
-require 'jwt'
 
 class Hyrax::Hirmeos::Client
-  attr_accessor :username, :password, :metrics_base_url, :translation_base_url, :secret
+  attr_accessor :username, :password, :metrics_base_url, :translation_base_url,
+  :token_base_url, :secret
 
-  def initialize(username, password, metrics_base_url, translation_base_url, secret)
+  def initialize(username, password, metrics_base_url, translation_base_url, token_base_url, secret)
```

Comment:
This many arguments seems like a code-smell. The Rubocop rule is there to try and guide you to better coding practices and make you think about how you are structuring your objects. 

The most simple solution would be to replace your named arguments here with a hash of values. 

```ruby
def initialize(args = {})
    @username = args[:username]
    @password = args[:password]
    @metrics_base_url = args[:metrics_base_url]
    @translation_base_url = args[:translation_base_url]
    @token_base_url = args[:token_base_url]
    @secret = args[:secret]
  end
```

```ruby 
arg = {
    username: username, 
    password: pasword, 
    metrics_base_url: metrics_base_url,
    translation_base_url: translation_base_url, 
    token_base_url: token_base_url,
    secret: secret
}
Hyrax::Hirmeos::Client.new(args)
```

Extracting the hash keys into instance variables can also be streamlined, for instance with something like this: 

```ruby
2.7.4 :002 > hsh = { one: 1, two: 2 }
 => {:one=>1, :two=>2}
2.7.4 :003 > @one, @two = hsh.values_at(:one, :two)
 => [1, 2]
2.7.4 :004 > @one
 => 1
2.7.4 :005 > @two
 => 2
```

But when you have a lot of different arguments its going to be another long and complex looking line. So we could iterate over the hash: 

```ruby
2.7.4 :006 > hsh.each { |k,v| instance_variable_set("@#{k}", v) }
 => {:one=>1, :two=>2}
2.7.4 :007 > @one
 => 1
2.7.4 :008 > @two
 => 2
```

This would appease rubocop, but doesn't really change the issue that 6 different values are required inorder to create an instance of the `Client`. 

When we look at the values we have, there are 2 different types, credentials and URLs: 

- Credentials
  - username
  - password
  - secret

- URLs
  - metrics_base_url
  - translation_base_url
  - token_base_url
  
So a possible solution would be to create a new `Credentials` object, which could hold the credentials from above and a `MetricsUrls` class to hold the base URLs. 

These could then be passed into `Client`, and their values accessed as required. 

```ruby
class Credentials
   attr_reader :username, :password, :secret
   def initalize(username, password, secret)
      @username = username
      #...
   end
end

class Client
   def initialize(credentials)
      @credentials = credentials
    end

    def request_token
       json = { email: @credentials.username, password: @credentials.password }.to_json
       response = Faraday.post(URI.join(token_base_url, 'tokens'), json)
    end
end

credentials = Credentials.new("foo@bar.com" ..., ...)
Hyrax::Hirmeos::Client.new(credentials, ...)
```
With this method you have broken down the concerns of the objects so that they are just doing a single thing and at the same time made it much more obvious what your `Client` requires.